audio/out/internal.h

```diff
@@ -201,6 +201,8 @@ struct ao_driver {
 // These functions can be called by AOs.
 
 int ao_read_data(struct ao *ao, void **data, int samples, int64_t out_time_ns);
+__attribute__((warn_unused_result))
```

Comment:
I think nearly all the `__attribute` usage is currently guarded via ifdefs or similar. I'd add `MP_WARN_UNUSED` or something similar to `osdep/compiler.h`.

```diff
--- a/osdep/compiler.h
+++ b/osdep/compiler.h
@@ -7,10 +7,12 @@
 #define PRINTF_ATTRIBUTE(a1, a2) __attribute__ ((format(printf, a1, a2)))
 #define MP_NORETURN __attribute__((noreturn))
 #define MP_FALLTHROUGH __attribute__((fallthrough))
+#define MP_WARN_UNUSED __attribute__((warn_unused_result))
 #else
 #define PRINTF_ATTRIBUTE(a1, a2)
 #define MP_NORETURN
 #define MP_FALLTHROUGH do {} while (0)
+#define MP_WARN_UNUSED
 #endif
```

Reply:
Not all of them:
```
ta/ta.h
24:#define TA_PRF(a1, a2) __attribute__ ((format(printf, a1, a2)))
34:#define TA_PRF(a1, a2) __attribute__ ((format (gnu_printf, a1, a2)))
libmpv/client.h
33:#define MPV_EXPORT __attribute__((visibility("default")))
```

And a few more. If it is preferred I can change it though.

Reply:
Both the example is under ifdefs and has a fallback though (which was my main point, doing it in `compiler.h` isn't necessary but it makes more sense than inlining it).

https://github.com/mpv-player/mpv/blob/84de84b8aa6023efb3f8fac368d802359657268c/ta/ta.h#L23-L29

https://github.com/mpv-player/mpv/blob/84de84b8aa6023efb3f8fac368d802359657268c/libmpv/client.h#L29-L38

Reply:
Indeed, update!